### PR TITLE
Feature/log computational graph

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/2915
Automatically logs the computational graph for certain loggers (tensorboard and test tube, could not find any other  loggers that is able to do this). Only requirement is that user has defined the `model.example_input_array` attribute which we already use for model summaries. 

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/models/test_cpu.py [hunk 1]
```diff
@@ -494,6 +496,7 @@ def train_dataloader(self):
     )
 
     model = BpttTestModel(**hparams)
+    model.example_input_array = None
```
Comment (justusschock) [inline tests/models/test_cpu.py:499]: can we also add a test here, where this is set? TO test that this actually works?

File: pytorch_lightning/loggers/base.py [hunk 2]
```diff
@@ -296,6 +306,10 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
         for logger in self._logger_iterable:
             logger.log_hyperparams(params)
 
+    def log_graph(self, model: LightningModule) -> None:
```
Comment (justusschock) [inline pytorch_lightning/loggers/base.py:309]: can we add an optional input argument here, which defaults to the `example_input_array` if None?

File: tests/models/test_cpu.py (hunk 1 above)
Comment (SkafteNicki) [reply tests/models/test_cpu.py]: Done :)

File: pytorch_lightning/loggers/base.py (hunk 2 above)
Comment (SkafteNicki) [reply pytorch_lightning/loggers/base.py]: Done :)

**Pull request merged**